src/nvim/main.c

```diff
@@ -145,6 +145,21 @@ void early_init(void)
 {
   handle_init();
 
+  // Use $NVIM and $NVIMRUNTIME as $VIM and $VIMRUNTIME
+  // or unset them to avoid accidentaly using the Vim runtime
+  const char *env_nvimruntime = os_getenv("NVIMRUNTIME");
+  if (env_nvimruntime) {
+    os_setenv("VIMRUNTIME", env_nvimruntime, true);
```

Comment:
Instead of this let's just replace the references to VIMRUNTIME in the nvim source (for reasons mentioned in the other comment). It's a bit more work, but it's the right thing to do.